stock_helper/tests/test_location_source_from_route.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+
+from .common import StockHelperCommonCase
+
+
+class TestStockLocationSoruceFromRoute(StockHelperCommonCase):
+    def test_get_source_location_from_route(self):
+        self.env["res.config.settings"].write(
+            {
+                "group_stock_adv_location": True,
+            }
+        )
+        self.wh.delivery_steps = "pick_pack_ship"
+        route = self.wh.delivery_route_id
```

Comment:
```suggestion
        route = self.wh.delivery_route_id
        delivery_route_3 = self.wh.delivery_route_id
        delivery_route_3.rule_ids[0].write({
            'location_dest_id': delivery_route_3.rule_ids[1].location_src_id.id,
        })
        delivery_route_3.rule_ids[1].write({'action': 'pull'})
        delivery_route_3.rule_ids[2].write({'action': 'pull'})
```
This will fix the error you get when running the test. so please restore the part you deleted
see: https://github.com/odoo/odoo/commit/11e69870db1c49d9a6af79ffd263e4e162b34b6b#diff-9571cf2bab8ee3cfa92d1d1a9c5bd9ead5be0adf66e8ab0e4596fe26f0e45833R11-R21